Projects-2017-2018/StringSortingLSD-MSDbyBuseDragomirAlexandru/LSD MSD string sorts/key_indexed_counting.cpp

```diff
@@ -0,0 +1,49 @@
+#include <iostream>
+#include <vector>
+#include <algorithm>
+#include <string.h>
+#include "Element.h"
+#include "key_indexed_counting.h"
+
+void key_indexed_counting(vector<Element> my_array, int alphabet_size)
+{
+    int counter[alphabet_size + 1] = {0};
+
+    for(auto it = my_array.begin(); it != my_array.end(); it ++)
+    {
+        counter[it -> getKey() + 1] ++;
+    }
+
+    for(int i = 0; i < alphabet_size; i ++)
+    {
+        counter[i + 1] = counter[i] + counter[i + 1];
+    }
+
+    Element *auxiliary_array = new Element[my_array.size()];
+
+    for(unsigned int index = 0; index < my_array.size(); index ++)
+    {
+        string current_word = my_array[index].getWord();
+        int current_key = my_array[index].getKey();
+        int current_position = counter[current_key];
+
+        auxiliary_array[current_position].setWord(current_word);
+        auxiliary_array[current_position].setKey(current_key);
+
+        counter[current_key] ++;
+    }
+
+    int dimension = my_array.size();

```

Comment:
move this line before for(unsigned int index = 0; index < my_array.size(); index ++) and use dimension in this for also.